src/app/ReadClient.cpp

```diff
@@ -828,6 +828,29 @@ void ReadClient::OnLivenessTimeoutCallback(System::Layer * apSystemLayer, void *
                  "Subscription Liveness timeout with SubscriptionID = 0x%08" PRIx32 ", Peer = %02x:" ChipLogFormatX64,
                  _this->mSubscriptionId, _this->GetFabricIndex(), ChipLogValueX64(_this->GetPeerNodeId()));
 
+    // We didn't get a message from the server on time; it's possible that it no
+    // longer has a useful CASE session to us.  Mark defunct all sessions that
+    // have not seen peer activity in at least as long as our session.
+    const auto & holder = _this->mReadPrepareParams.mSessionHolder;
+    if (holder)
```

Comment:
Out of curiosity, in which scenarios is it useful to loop over all sessions instead of just do `holder->MarkAsDefunct`? Do clients try establish another session to a node without marking the previous one as defunct?

Reply:
@bzbarsky-apple is OoO, but filed an issue for this. He'll follow up!

Reply:
https://github.com/project-chip/connectedhomeip/issues/21896#issuecomment-1218287800 has the answer I was going to type.  ;)